parcel.go

```diff
@@ -13,48 +15,123 @@ func NewParcelStore(db *sql.DB) ParcelStore {
 }
 
 func (s ParcelStore) Add(p Parcel) (int, error) {
-	// реализуйте добавление строки в таблицу parcel, используйте данные из переменной p
 
-	// верните идентификатор последней добавленной записи
-	return 0, nil
+	db, err := sql.Open("sqlite", "tracker.db")
+	if err != nil {
+		return 0, err
+	}
+	defer db.Close()
+
+	res, err := db.Exec("insert into parcel (client, status, address, created_at) values (:client, :status, :address, :created_at)",
+		sql.Named("client", p.Client),
+		sql.Named("status", p.Status),
+		sql.Named("address", p.Address),
+		sql.Named("created_at", p.CreatedAt),
+	)
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := res.LastInsertId()
+	if err != nil {
+		return 0, err
+	}
+
+	return int(l), nil
 }
 
 func (s ParcelStore) Get(number int) (Parcel, error) {
-	// реализуйте чтение строки по заданному number
-	// здесь из таблицы должна вернуться только одна строка
-
-	// заполните объект Parcel данными из таблицы
 	p := Parcel{}
 
+	db, err := sql.Open("sqlite", "tracker.db")
+	if err != nil {
+		return p, err
+	}
+	defer db.Close()
+
+	row := db.QueryRow("select * from parcel where number = :number", sql.Named("number", number))
+
+	err = row.Scan(&p.Number, &p.Client, &p.Status, &p.Address, &p.CreatedAt)
+	if err != nil {
+		return p, err
+	}
+
 	return p, nil
 }
 
 func (s ParcelStore) GetByClient(client int) ([]Parcel, error) {
-	// реализуйте чтение строк из таблицы parcel по заданному client
-	// здесь из таблицы может вернуться несколько строк
-
-	// заполните срез Parcel данными из таблицы
 	var res []Parcel
 
+	db, err := sql.Open("sqlite", "tracker.db")
+	if err != nil {
+		return res, err
+	}
+	defer db.Close()
+
+	rows, err := db.Query("select * from parcel where client = :client", sql.Named("client", client))
+	if err != nil {
+		return res, err
```

Comment:
При ошибке лучше возвращать `nil`, чтобы не было полуданных
```suggestion
		return nil, err
```

Reply:
о, круто!
не думал, что можно возвращать nil вместо слайса

а что можно вернуть вместо кастомной структуры (когда на выходе ожидается Parcel)?

Reply:
Parcel{}